srecord/memory/walker/adler16.h

```diff
@@ -34,27 +34,11 @@ class memory_walker_adler16:
     public memory_walker
 {
 public:
-    typedef std::shared_ptr<memory_walker_adler16> pointer;
-
     /**
       * The destructor.
       */
     ~memory_walker_adler16() override = default;
 
-private:
```

Comment:
This inclusion of private here is intentional. The idea is that the headers are written from a template and in a very structured way. Rather than mysteriously move the constructor to the bottom of the header, it's left in the usual spot as a prompt (to the lib user especially) that the create method should be used.